src/components/messages-view-configurator/MessagesViewConfigurator.tsx

```diff
@@ -32,11 +32,17 @@ const MessageViewConfigurator = ({ sessions }: Props) => {
 
 	const modalRef = useRef<HTMLDivElement>(null);
 	const buttonRef = useRef<HTMLButtonElement>(null);
+	const imageRef = useRef<HTMLImageElement>(null);
 
 	useOutsideClickListener(modalRef, (e: MouseEvent) => {
 		const isFromAutocomplete = Boolean((e.target as HTMLElement).closest('.rules-autocomplete'));
 		const isFromSelect = Boolean((e.target as HTMLElement).closest('.rules-select-options-list'));
-		if (e.target !== buttonRef.current && !isFromAutocomplete && !isFromSelect) {
+		if (
+			e.target !== buttonRef.current &&
+			e.target !== imageRef.current &&
```

Comment:
instead of having 2 refs and checking every you can replace it with rootRef.contains(e.target)